### PR TITLE
Add typings for index.coffee

### PR DESCRIPTION
**Pull request by CameronDiver**
Doesn't contain types for legacy code.

Change-type: minor
Connects-to: #36
Signed-off-by: Cameron Diver <cameron@resin.io>
---- Autogenerated Waffleboard Connection: Connects to #36

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "url": "git@github.com:resin-io/docker-progress.git"
   },
   "author": "Pagan Gazzard <page@resin.io>",
+	"types": "index.d.ts",
```
Comment (dfunckt) [inline package.json]: This file seems to be using spaces instead of tabs.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+declare module 'docker-progress' {
+
+	type ProgressCallback = (progressObj: any) => void;
+
+	class DockerProgress {
+
+		public constructor(opts: any);
```
Comment (dfunckt) [inline index.d.ts]: `opts` are actually optional -- should this be `opts?: any` instead?

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+		public aggregateProgress(count: number, onProgress: ProgressCallback): ProgressCallback[];
+
+		public pull(
+			image: string,
+			onProgress: ProgressCallback,
+			options: any,
+			callback: ((err: Error, output: string) => void) | null,
```
Comment (dfunckt) [inline index.d.ts]: Isn't this better written as `callback?: (err: Error, output: string) => void`?

File: package.json (hunk 1 above)
Comment (CameronDiver) [reply package.json]: God damn it vim...

File: index.d.ts (hunk 3 above)
Comment (dfunckt) [reply index.d.ts]: Actually, wait -- the callback is actually `callback?: (err: Error) => void`

**Review by dfunckt — APPROVED** [review]

**Pull request merged**